presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -105,6 +112,16 @@ private PrestoDriverUri(URI uri, Properties driverProperties)
         address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
         properties = mergeConnectionProperties(uri, driverProperties);
 
+        // Remove the default SSL_VERIFICATION property in the following cases in which SSL_VERIFICATION is not allowed
```

Comment:
I find removing properties hacky, I would prefer to modify logic below instead to handle mentioned cases.

Reply:
Instead of removing the default `SSL_VERIFICATION` property in specific cases, I added a `isProvided` variable in `AbstractConnectionProperty` which is used to compose a `Predicate` to determine whether the `SSL_VERIFICATION` property is allowed or not.